src/core/imported/addrlib/src/core/addrlib2.cpp

```diff
@@ -1159,7 +1159,8 @@ ADDR_E_RETURNCODE Lib::ComputeSurfaceAddrFromCoordLinear(
      ) const
 {
     ADDR_E_RETURNCODE returnCode = ADDR_OK;
-    BOOL_32 valid = (pIn->numSamples <= 1) && (pIn->numFrags <= 1) && (pIn->pipeBankXor == 0);
+    BOOL_32 valid = (pIn->numSamples <= 1) && (pIn->numFrags <= 1) && (pIn->pipeBankXor == 0) &&
+                    (pIn->numMipLevels <= MaxMipLevels);
```

Comment:
This fix looks fine, though it's only a partial fix -- it wouldn't cover tiled Images.  To correct this for both tiled _and_ linear Images, you'd want to fix it inside `Valid3DMipSliceIdConstraint`.  See the call site here: https://github.com/GPUOpen-Drivers/pal/blob/a40241a1e6f5970a9313efe686a79c0a429ea5eb/src/core/imported/addrlib/src/core/addrlib2.cpp#L367)